### PR TITLE
[MU4] Implemented autosave for new project

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #10392

### CODE REVIEW

File: src/project/internal/notationproject.cpp [hunk 1]
```diff
@@ -98,6 +98,11 @@ static void setupScoreMetaTags(Ms::MasterScore* masterScore, const ProjectCreate
     }
 }
 
+QString scoreDefaultTitle()
```
Comment (igorkorsukov) [inline src/project/internal/notationproject.cpp:101]: better add `static`

File: src/project/internal/notationproject.cpp [hunk 2]
```diff
@@ -140,13 +145,7 @@ mu::Ret NotationProject::load(const io::path& path, const io::path& stylePath, b
...
 
     MscReader::Params params;
-
-    bool needRestoreUnsavedChanges = projectAutoSaver()->projectHasUnsavedChanges(path);
-    if (needRestoreUnsavedChanges) {
-        params.filePath = projectAutoSaver()->projectAutoSavePath(path).toQString();
-    } else {
-        params.filePath = path.toQString();
-    }
+    params.filePath = autosavePath.empty() ? path.toQString() : autosavePath.toQString();
```
Comment (igorkorsukov) [inline src/project/internal/notationproject.cpp:148]: Can we move this logic to a higher level? (where called)

File: src/framework/global/io/path.cpp [hunk 3]
```diff
@@ -110,6 +110,12 @@ mu::io::path mu::io::basename(const mu::io::path& path)
     return fi.baseName();
 }
 
+mu::io::path mu::io::complateBasename(const mu::io::path& path)
```
Comment (cbjeukendrup) [inline src/framework/global/io/path.cpp:113]: Typo: `complate` should be `complete` :)

**Comment by DmitryArefiev** [comment]
Tested #10392 on Win10, Mac12, LinuxMint 20.2 - **FIXED**

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**